pkg/docker/util.go

```diff
@@ -54,27 +64,88 @@ func GetImageRegistryAuth(auths *client.AuthConfigurations, imageName string) cl
 		glog.V(5).Infof("Using %s credentials for pulling %s", auth.Email, imageName)
 		return auth
 	}
-	return client.AuthConfiguration{}
+	return api.AuthConfig{}
 }
 
 // LoadImageRegistryAuth loads and returns the set of client auth objects from a docker config
 // json file.
-func LoadImageRegistryAuth(dockerCfg io.Reader) *client.AuthConfigurations {
-	auths, err := client.NewAuthConfigurations(dockerCfg)
+func LoadImageRegistryAuth(dockerCfg io.Reader) *AuthConfigurations {
+	auths, err := NewAuthConfigurations(dockerCfg)
 	if err != nil {
 		glog.V(0).Infof("error: Unable to load docker config")
 		return nil
 	}
 	return auths
 }
 
+// begin borrowed from fsouza
```

Comment:
the k8s stuff also pulls from well known env vars ... it was my thinking that those env vars get set when the various openshift components run, but that doesn't apply for s2i